HADOOP-18724. Open file fails with NumberFormatException for S3AFileSystem

steveloughran opened the pull request:
+s3a openfile parser downgrades on all parser failures.

No tests for this yet, this is up to get the API reviewed


### How was this patch tested?


### For code changes:

- [X] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

steveloughran committed:
HADOOP-18724. map FSbuilder float/double to the new long setter

+s3a openfile parser downgrades on all parser failures.

Change-Id: I3fc2048c19ceb80215a8d517aa355794746379b4

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 35s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  17m 21s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  26m  0s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  23m 11s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  compile  |  20m 31s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  checkstyle  |   3m 46s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 43s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 54s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   1m 38s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |   4m  4s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  21m 35s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 33s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  22m 20s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| -1 :x: |  javac  |  22m 20s | [/results-compile-javac-root-jdkUbuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/1/artifact/out/results-compile-javac-root-jdkUbuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1.txt) |  root-jdkUbuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 generated 3 new + 2806 unchanged - 0 fixed = 2809 total (was 2806)  |
| +1 :green_heart: |  compile  |  20m 32s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| -1 :x: |  javac  |  20m 32s | [/results-compile-javac-root-jdkPrivateBuild-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/1/artifact/out/results-compile-javac-root-jdkPrivateBuild-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09.txt) |  root-jdkPrivateBuild-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 generated 3 new + 2604 unchanged - 0 fixed = 2607 total (was 2604)  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   3m 35s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   2m 41s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 47s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   1m 24s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |   4m 10s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 34s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  18m 39s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 11s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  4s |  |  The patch does not generate ASF License warnings.  |
|  |   | 231m  7s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5611 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 0a94e1b08367 4.15.0-206-generic #217-Ubuntu SMP Fri Feb 3 19:10:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 54ce14d8a71d4b16c702bb78781fdaced9c91195 |
| Default Java | Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/1/testReport/ |
| Max. process+thread count | 1999 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran committed:
HADOOP-18724. suppress deprecation; update javadocs

Change-Id: I0925fcb76e84a462b17403411402f1bcd5af24dc

steveloughran committed:
HADOOP-18724. testing and improve use

* New FSBuilderSupport class to do resilient parsing &c
* Test for this
* Update docs, including with history

Change-Id: I223d11b8a0998bd4c3f10690d45d482b12f6a3e7

steveloughran commented:
testing
* azure cardiff:  -Dparallel-tests=abfs -DtestsThreadCount=6
* s3 london:  -Dparallel-tests -DtestsThreadCount=8

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 36s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 5 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  16m 14s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  19m 35s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  15m 44s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  compile  |  14m 20s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  checkstyle  |   3m 45s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   5m  8s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   4m 22s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m  2s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |   8m  7s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  21m 12s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 43s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  14m 56s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javac  |  14m 56s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  14m 21s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  javac  |  14m 21s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/2/artifact/out/blanks-eol.txt) |  The patch has 2 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -0 :warning: |  checkstyle  |   3m 36s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/2/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 351 unchanged - 0 fixed = 353 total (was 351)  |
| +1 :green_heart: |  mvnsite  |   5m  5s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m  6s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m  1s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |   8m 53s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 33s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  18m 20s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 42s |  |  hadoop-yarn-common in the patch passed.  |
| +1 :green_heart: |  unit  |   7m  0s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  8s |  |  hadoop-mapreduce-examples in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 40s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  3s |  |  The patch does not generate ASF License warnings.  |
|  |   | 234m 30s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5611 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets markdownlint |
| uname | Linux 0d7490c92bcb 4.15.0-206-generic #217-Ubuntu SMP Fri Feb 3 19:10:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / aaf35ac212f0c048849ab9d97ed70c220b9afbc3 |
| Default Java | Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/2/testReport/ |
| Max. process+thread count | 2524 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn reviewed (commented):
Some minor Jenkins warnings that need to fixed, else changes looks good

steveloughran committed:
HADOOP-18724. checkstyle, javadoc, test enhancements

Change-Id: Iec77133e97352c4d36a3b176903a700d7b0065bc

steveloughran committed:
HADOOP-18724. azure stores to recognize all fs.openfile keys

Change-Id: Ibc33b7ca11cff7522fbe49ff23ce80165f1de04f

steveloughran committed:
HADOOP-18724. Pull test up.

move the test with a floating point len from hadoop-aws to hadoop-common,
so verify that all filesystems can cope with it.

Change-Id: Ie982f972f59f7a3547551592adfd8541ce248f71

steveloughran commented:
* fixed the yetus complaints
* verified that a floating point drain threshold is ok
* fixed up abfs so it doesn't reject any must() arg of the "well known fs.option.openfile" values, even though it doesn't (yet) process them

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m 25s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 7 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  22m 29s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  19m 38s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  15m 41s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  compile  |  14m 26s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  checkstyle  |   3m 50s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   6m  2s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   5m  8s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m 44s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |   9m 19s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  21m 11s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 12s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  14m 59s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javac  |  14m 59s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m  9s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  javac  |  17m  9s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   4m  0s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 373 unchanged - 0 fixed = 375 total (was 373)  |
| +1 :green_heart: |  mvnsite  |   5m 39s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 30s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m 16s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |  10m 48s |  |  the patch passed  |
| -1 :x: |  shadedclient  |  29m 37s |  |  patch has errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   5m 11s | [/patch-unit-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt) |  hadoop-common in the patch failed.  |
| -1 :x: |  unit  |   0m 40s | [/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-common.txt) |  hadoop-yarn-common in the patch failed.  |
| -1 :x: |  unit  |   0m 41s | [/patch-unit-hadoop-mapreduce-project_hadoop-mapreduce-client_hadoop-mapreduce-client-core.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/artifact/out/patch-unit-hadoop-mapreduce-project_hadoop-mapreduce-client_hadoop-mapreduce-client-core.txt) |  hadoop-mapreduce-client-core in the patch failed.  |
| -1 :x: |  unit  |   0m 40s | [/patch-unit-hadoop-mapreduce-project_hadoop-mapreduce-examples.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/artifact/out/patch-unit-hadoop-mapreduce-project_hadoop-mapreduce-examples.txt) |  hadoop-mapreduce-examples in the patch failed.  |
| -1 :x: |  unit  |   0m 41s | [/patch-unit-hadoop-tools_hadoop-aws.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/artifact/out/patch-unit-hadoop-tools_hadoop-aws.txt) |  hadoop-aws in the patch failed.  |
| -1 :x: |  unit  |   0m 40s | [/patch-unit-hadoop-tools_hadoop-azure.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/artifact/out/patch-unit-hadoop-tools_hadoop-azure.txt) |  hadoop-azure in the patch failed.  |
| +0 :ok: |  asflicense  |   0m 41s |  |  ASF License check generated no output?  |
|  |   | 240m 28s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ipc.TestServer |
|   | hadoop.security.TestUGILoginFromKeytab |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5611 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets markdownlint |
| uname | Linux a4e24e120c10 4.15.0-206-generic #217-Ubuntu SMP Fri Feb 3 19:10:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 7aae2cf57cc60aa5791403541b00152ac164ad7a |
| Default Java | Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/testReport/ |
| Max. process+thread count | 622 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-tools/hadoop-aws hadoop-tools/hadoop-azure U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   2m 22s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 7 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  24m 31s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  22m  7s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  17m 29s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  compile  |  16m  7s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  checkstyle  |   4m  6s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   5m 13s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   4m 11s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   3m 44s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |   9m  4s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  23m 52s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 10s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 41s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javac  |  16m 41s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 49s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  javac  |  15m 49s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   3m 56s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/4/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 373 unchanged - 0 fixed = 375 total (was 373)  |
| +1 :green_heart: |  mvnsite  |   5m  8s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m  7s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m 24s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |  11m  1s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  24m 58s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  18m 32s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 20s |  |  hadoop-yarn-common in the patch passed.  |
| +1 :green_heart: |  unit  |   6m 53s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 57s |  |  hadoop-mapreduce-examples in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 25s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 12s |  |  hadoop-azure in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 53s |  |  The patch does not generate ASF License warnings.  |
|  |   | 266m 31s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5611 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets markdownlint |
| uname | Linux ed659d8b11b3 4.15.0-206-generic #217-Ubuntu SMP Fri Feb 3 19:10:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 7aae2cf57cc60aa5791403541b00152ac164ad7a |
| Default Java | Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/4/testReport/ |
| Max. process+thread count | 3101 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-tools/hadoop-aws hadoop-tools/hadoop-azure U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/4/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  0s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 7 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  18m 31s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  20m 19s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  16m 32s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  compile  |  15m 20s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  checkstyle  |   3m 53s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   5m 42s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   4m 49s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m 47s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |   9m 19s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  22m 13s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 18s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 39s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javac  |  15m 39s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 14s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  javac  |  15m 14s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   3m 46s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/5/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 2 new + 373 unchanged - 0 fixed = 375 total (was 373)  |
| +1 :green_heart: |  mvnsite  |   5m 32s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 21s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m 26s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |  10m 26s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  23m 52s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  18m 44s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 43s |  |  hadoop-yarn-common in the patch passed.  |
| +1 :green_heart: |  unit  |   7m 16s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  9s |  |  hadoop-mapreduce-examples in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 51s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 18s |  |  hadoop-azure in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 57s |  |  The patch does not generate ASF License warnings.  |
|  |   | 260m  9s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5611 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets markdownlint |
| uname | Linux 64f44e9ff303 4.15.0-206-generic #217-Ubuntu SMP Fri Feb 3 19:10:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 7aae2cf57cc60aa5791403541b00152ac164ad7a |
| Default Java | Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/5/testReport/ |
| Max. process+thread count | 1597 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-tools/hadoop-aws hadoop-tools/hadoop-azure U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/5/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn reviewed (approved):
Changes LGTM.
Checkstyle warning regarding unused import. Commit once fixed..

steveloughran committed:
HADOOP-18724. checkstyle

Change-Id: I4be5411aa933823386444c046f6927e55c2b27fe

steveloughran commented:
ok, pushed up a pr with the fixed imports; if yetus is happy will merge and then backport

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 7 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 46s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  19m 27s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  15m 42s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  compile  |  14m 32s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  checkstyle  |   3m 43s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   6m  4s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   5m  9s |  |  trunk passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m 48s |  |  trunk passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |   9m 16s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  20m 58s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 13s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  14m 59s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javac  |  14m 59s |  |  the patch passed  |
| -1 :x: |  compile  |  10m 19s | [/patch-compile-root-jdkPrivateBuild-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/6/artifact/out/patch-compile-root-jdkPrivateBuild-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09.txt) |  root in the patch failed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09.  |
| -1 :x: |  javac  |  10m 19s | [/patch-compile-root-jdkPrivateBuild-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/6/artifact/out/patch-compile-root-jdkPrivateBuild-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09.txt) |  root in the patch failed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   3m 39s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   5m 57s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   4m 57s |  |  the patch passed with JDK Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1  |
| +1 :green_heart: |  javadoc  |   4m 39s |  |  the patch passed with JDK Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09  |
| +1 :green_heart: |  spotbugs  |  10m 12s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  22m 34s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  18m 37s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 42s |  |  hadoop-yarn-common in the patch passed.  |
| +1 :green_heart: |  unit  |   6m 59s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  8s |  |  hadoop-mapreduce-examples in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 43s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 17s |  |  hadoop-azure in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  2s |  |  The patch does not generate ASF License warnings.  |
|  |   | 241m 16s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5611 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets markdownlint |
| uname | Linux e6105507200a 4.15.0-206-generic #217-Ubuntu SMP Fri Feb 3 19:10:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / d2fabe9ae3f3b21250709729caaabcd44be0c3f1 |
| Default Java | Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.18+10-post-Ubuntu-0ubuntu120.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_362-8u362-ga-0ubuntu1~20.04.1-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/6/testReport/ |
| Max. process+thread count | 1923 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-tools/hadoop-aws hadoop-tools/hadoop-azure U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5611/6/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
yarn bower build failed with network issues; 100% unrelated
